mypy/binder.py

```diff
@@ -222,6 +226,12 @@ def pop_frame(self, can_skip: bool, fall_through: int) -> Frame:
 
     @contextmanager
     def accumulate_type_assignments(self) -> 'Iterator[Assigns]':
+        """Push a new map to collect assigned types in multiassign from union.
+
+        If this map is not None, actual binding is deferred until all items in
+        the union are processed (a union of collected items is later bound
+        manually by the caller).
+        """
         self.type_assignments = defaultdict(list)
         yield self.type_assignments
         self.type_assignments = None
```

Comment:
What about things like `(x, (y, z)) = ...` where there are two levels of union type multiple assignment? Would we need to restore old mapping perhaps? This would be worth testing as well.

Reply:
Yes, you are totally right. Will push a commit in a second. Also testing this uncovered another flaw in my implementation: I need to iterate over nested lvalues in ``for`` loops (added a test for this as well).

Reply:
Done!